### PR TITLE
PLT-5547 #comment PH APIs integrated with QuarticSDK

### PR DESCRIPTION
**Pull request by HimanshuS24**
Jira: https://quarticai.atlassian.net/browse/PLT-5547

This PR contains changes for the integration of Product Harbor APIs that includes the following:-

1. Implementation of Site, Product, Procedure, and ProcedureStep Entities.
2. Implementation of the logic for fetching and creation of Product-->Procedures.
3. Implementation of the logic for fetching and creation of 
Procedure-->ProcedureSteps(UnitProcedure/Operation/Phase/PhaseStep)
4. Documentation Added

Though this PR contains changes in 130 files but most of the files/changes are auto-generated by Sphinx. **The main file to review is python file changes.** 

PS: Corresponding [Deming](https://github.com/Quarticai/deming-core/pull/986/) and [ContexAlyze](https://github.com/Quarticai/ContexAlyze/pull/3990) PR are in review.

### CODE REVIEW

File: quartic_sdk/utilities/constants.py [hunk 1]
```diff
@@ -151,15 +160,42 @@
...
     SAFETY: 'Safety',
-    ENVIRONMENT: 'Environment'
+    ENVIRONMENT: 'Environment',
+    OTHERS: 'Others'
 }
 
 DEFAULT_PAGE_LIMIT_ROWS = 200000
 DEFAULT_BATCH_SIZE = 200000
+
+# Step Types in ProcedureStep hierarchy
```
Comment (scriperdjq) [inline quartic_sdk/utilities/constants.py:178]: why do we need this? we could have enum classes instead

File: quartic_sdk/core/entities/site.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    returned by the API
+    """
+
+    def __repr__(self):
+        """
+        Override the method to return the site name
+        """
+        return f"<{Constants.SITE_ENTITY}: {self.name}>"
+
+    def assets(self):
```
Comment (scriperdjq) [inline quartic_sdk/core/entities/site.py:20]: are we planning to implement these?

File: quartic_sdk/utilities/utils.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        self.duration_ms = duration_ms
+
+    def raw_json(self):
+        """
+        This method returns raw_json schema required for RuleDefinition creation
+        :return: Json Schema
+        """
+        return {
+            "0": {
+                Constants.TAG: str(self.first_tag.id)
```
Comment (scriperdjq) [inline quartic_sdk/utilities/utils.py:35]: define constants within class if there is no used outside

File: quartic_sdk/core/entities/site.py (hunk 2 above)
Comment (HimanshuS24) [reply quartic_sdk/core/entities/site.py:20]: No, we are not planning to implement this yet. But can be taken up in the future if needed. 
cc. @mprasoonquartic

File: quartic_sdk/utilities/constants.py (hunk 1 above)
Comment (HimanshuS24) [reply quartic_sdk/utilities/constants.py:178]: actually only `UNIT_PROCEDURE` is needed. I will move this constant to class specifically using this

File: quartic_sdk/utilities/utils.py (hunk 3 above)
Comment (HimanshuS24) [reply quartic_sdk/utilities/utils.py:35]: sure

File: quartic_sdk/core/entities/site.py (hunk 2 above)
Comment (mprasoonquartic) [reply quartic_sdk/core/entities/site.py:20]: Yes. an expected behavior with sites would be to fetch its assets etc, which we are not implementing right now. Thus, proactively adding NotImplementedError to handle the case later

**Review by mprasoonquartic — APPROVED** [review]

**Review by scriperdjq — APPROVED** [review]
@mprasoonquartic @shivinquartic could find some way to generate the docs only during build & not having to commit in git..we wont be reviewing the autogenerated files anyways

**Comment by shivinquartic** [comment]
> @mprasoonquartic @shivinquartic could find some way to generate the docs only during build & not having to commit in git..we wont be reviewing the autogenerated files anyways

Yeah, that how I thought of doing it in the new SDK.

**Pull request merged**